### PR TITLE
Milestone 7 Vet Clinic | Database schema diagram based

### PR DESCRIPTION
**Pull request by michael-duke**
## Vet Clinic milestone 7: Diagram-based Database schema (Clinic)

In this milestone @kendoriddy  and @michael-duke have:
- Followed the above `clinic diagram` provided to create the `clinic` schema in the `schema_based_on_diagram.sql`
- Implemented Join tables from `many-to-many` relationships from the diagram,
   - Table `medical_histories_treatments` from `medical_histories` and `treatments`
- Rearranged tables correctly so it doesn't break when executing the .sql file
- Added `foreign keys` as advised, and learned about datatypes (`Timestamps`)
- Create `FK indexes` for better performance when queriying

image

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    quantity INT NOT NULL,
+    total_price DECIMAL NOT NULL,
+    invoice_id INT NULL REFERENCES invoices(id) ON DELETE CASCADE,
+    treatment_id INT NULL REFERENCES treatments(id) ON DELETE CASCADE
+);
+
+CREATE TABLE medical_histories_treatments (
+    medical_history_id INT REFERENCES medical_histories(id) ON DELETE CASCADE,
+    treatment_id INT REFERENCES treatments(id) ON DELETE CASCADE
+);
```
Comment (Wes-Isaac) [inline schema_based_on_diagram.sql]: - [x] Please add indexes for all foreign key columns for improved performance, as per [this](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.) requirement.

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @michael-duke & @kendoriddy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (michael-duke) [reply schema_based_on_diagram.sql]: Thanks @Wes-Isaac for your feeback, happy coding :rocket:

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @michael-duke 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by michael-duke** [comment]
Thanks for your feedback @youmari happy coding 😄 🚀

**Pull request merged**